main.go

```diff
@@ -46,48 +68,116 @@ func main() {
 		return
 	}
 
-	duration, err := config.getInterval()
-	if err != nil {
-		log.WithError(err).Error("Failed to parse interval")
-		return
-	}
-
 	if *dryRun {
-		render(config, clientset, tmpl, false)
+		render(config.OutTemplate, clientset, tmpl)
 	} else {
 		for range time.NewTicker(duration).C {
-			render(config, clientset, tmpl, true)
+			err = render(config.OutTemplate, clientset, tmpl)
+			if err != nil {
+				log.WithError(err).Error("Failed to render template")
+				opsStatus <- &OpsStatus{isSuccess: false, timestamp: time.Now(), error: err}
+				continue //we don't bother to exec hooks since the rendering failed
+			}
+			err = execHooks(config, opsStatus)
+			if err != nil {
+				opsStatus <- &OpsStatus{isSuccess: false, timestamp: time.Now(), error: err}
+				continue
+			}
+			opsStatus <- &OpsStatus{isSuccess: true, timestamp: time.Now()}
+			select {
+			case <-mainExit:
+				log.Info("Gracefully shutting down...")
+				log.Info("Waiting for server to shutdown...")
+				time.Sleep(5 * time.Second)
+				return
+			default:
+				continue
+			}
 		}
 	}
 }
+func buildHealthServer(status chan *OpsStatus, duration time.Duration, port uint32) *http.Server {
+	lastReport := OpsStatus{isSuccess: true, timestamp: time.Now()}
+	healthHandler := healthHandler{opsStatus: status, cacheExpirationTime: duration, lastReport: &lastReport}
+	http.HandleFunc("/health", healthHandler.ServeHTTP)
+	server := &http.Server{
+		Addr:    fmt.Sprintf(":%d", port),
+		Handler: http.DefaultServeMux,
+	}
+	return server
+}
+
+type healthHandler struct {
+	opsStatus           chan *OpsStatus
+	cacheExpirationTime time.Duration
+	lastReport          *OpsStatus
+	sync.Mutex
+}
 
-func bootstrapHealthCheck(port uint32) {
-	http.HandleFunc("/health", func(writer http.ResponseWriter, request *http.Request) {
-		fmt.Fprintf(writer, "I am alive !\n")
-	})
-	err := http.ListenAndServe(fmt.Sprintf(":%d", port), nil)
+func (hh healthHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
```

Comment:
related to previous comment - this function won't be needed if you run it in main.